src/Bounds.cpp

```diff
@@ -547,11 +547,21 @@ class Bounds : public IRVisitor {
 
     void visit(const Load *op) {
         op->index.accept(this);
-        if (interval.is_single_point()) {
-            // If the index is const we can return the load of that index
+        if (interval.is_single_point() && is_one(op->predicate)) {
+            // If the index is const and it is not a predicated load,
+            // we can return the load of that index
+            Expr scalar_pred = const_true();
+            if (const Call *c = op->predicate.as<Call>()) {
```

Comment:
It can either be Call::bool_to_mask or Call:: cast_mask. Theoretically, this shouldn't be possible since they are only introduced by EliminateBoolVectors and Bounds is never called after that pass right now. Should I just change it into an assertion that it can't be Call::bool_to_mask or Call::cast_mask?